### DISCUSSION TITLE
How to prevent slow updateTransforms?

### DISCUSSION THREAD

**Discussion by manvankaliber** [pixijs/pixijs#8255] [Q&A]
I am having trouble debugging my game's bad performance. 

My `stage` is a `pixi-viewport`. It has 1024 child `Container`'s, each having 64 tile child `Container`'s, that each contain two child `Sprite`'s.
The sprites use the same textures, one of which is tinted with a random color.

![image](https://user-images.githubusercontent.com/12383587/161393665-7745058c-d21f-43dc-a31f-3579241210cc.png)

_All brown squares you see without a label are also updateTransforms. Not sure why there are gaps here and there. The orange squares on the bottom right are graphics being rendered, flushed etc._

Why could `updateTransform` be taking so much time? I am not changing the positions of any of the objects ever. I am also not moving the viewport.

Can I prevent updateTransform from happening?

**Comment by miltoncandelero** [comment]
try removing the viewport, just raw containers. Does it get better? 🤔 
updateTransform shouldn't change if you didn't move anything 🤔

**Reply by manvankaliber** [reply]
> try removing the viewport, just raw containers. Does it get better? 🤔 updateTransform shouldn't change if you didn't move anything 🤔

It is also an issue without viewport. I truly am not moving anything ever...

**Reply by bigtimebuddy** [reply]
Can you please show your work or something that replicates the issue and not just the performance graph? That will help to see if we can reproduce.

**Comment by ivanpopelyshev** [comment]
tiles? use pixi-tilemap plugin

**Reply by manvankaliber** [reply]
It isn't really tiles.. Think more of them as grids with cells. I called them tiles to not cause any confusion.

**Reply by ivanpopelyshev** [reply]
you have to make custom mesh, or use pixi-tilemap, or just beginFill()'s in graphics. Either you group your objects, either transform update will be slow as heck )
